src/sentry/sentry_apps/tasks/sentry_apps.py

```diff
@@ -824,25 +823,34 @@ def send_webhooks(installation: RpcSentryAppInstallation, event: str, **kwargs:
         operation_type=SentryAppInteractionType.SEND_WEBHOOK,
         event_type=SentryAppEventType(event),
     ).capture() as lifecycle:
+        app_has_valid_webhook = (
+            installation.sentry_app.webhook_url is not None
+            and event in installation.sentry_app.events
+        )
+
         servicehook: ServiceHook | None = _load_service_hook(
             installation.organization_id, installation.id
         )
-        if not servicehook:
-            lifecycle.add_extra("events", installation.sentry_app.events)
-            lifecycle.add_extras(
-                {
-                    "installation_uuid": installation.uuid,
+        if not app_has_valid_webhook:
+            lifecycle.record_halt(SentryAppWebhookFailureReason.MISSING_SERVICEHOOK)
+            return
+
+        if not servicehook or (event not in servicehook.events):
+            logger.info(
+                "regenerating service hook for installation",
+                extra={
                     "installation_id": installation.id,
-                    "organization": installation.organization_id,
-                    "sentry_app": installation.sentry_app.id,
-                    "events": installation.sentry_app.events,
-                    "webhook_url": installation.sentry_app.webhook_url or "",
-                }
+                    "sentry_app_id": installation.sentry_app.id,
+                    "event": event,
+                    "servicehook_id": servicehook.id if servicehook else None,
+                },
             )
-            raise SentryAppSentryError(message=SentryAppWebhookFailureReason.MISSING_SERVICEHOOK)
-        if event not in servicehook.events:
-            raise SentryAppSentryError(
-                message=SentryAppWebhookFailureReason.EVENT_NOT_IN_SERVCEHOOK
+            hook_service.create_or_update_webhook_and_events_for_installation(
+                installation_id=installation.id,
+                organization_id=installation.organization_id,
+                webhook_url=installation.sentry_app.webhook_url,
+                events=installation.sentry_app.events,
+                application_id=installation.sentry_app.application_id,
```

Comment:
Is it possible this is `None`, and is that a concern? If the `ApiApplication` is deleted it will set `sentry_app.application` to null